src/convertToObject.js

```diff
@@ -7,6 +7,47 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const result = {};
+
+  let cleanedString = '';
+  let insideValue = false;
+
+  for (let i = 0; i < sourceString.length; i++) {
+    const char = sourceString[i];
+
+    if (insideValue) {
+      cleanedString += char;
+
+      if (char === ';') {
+        insideValue = false;
+      }
+    } else {
+      if (char === '\n' || char === '\t') {
+        continue;
+      } else if (char === ':') {
+        insideValue = true;
+      }
+
+      cleanedString += char;
+    }
+  }
+
+  const rules = cleanedString.split(';');
+
+  rules.forEach((rule) => {
+    const splitIndex = rule.indexOf(':');
+
+    if (splitIndex !== -1) {
+      const property = rule.slice(0, splitIndex).trim();
+      const value = rule.slice(splitIndex + 1).trim();
+
+      if (property && value) {
+        result[property] = value;
+      }
+    }
+  });
```

Comment:
[CODE STYLE] - All chained methods called on the same level should have the same indentation. Since you're using a loop instead of chained methods, this is a moot point, but if you refactor to use chained methods, keep this in mind for readability.